src/Dialect/Krnl/KrnlOps.td

```diff
@@ -205,3 +205,16 @@ def KrnlGlobalOp : Op<Krnl_Dialect, "global"> {
   let parser = ?;
   let printer = ?;
 }
+
+def KrnlGetRefOp : Op<Krnl_Dialect, "getref"> {
+  let summary = "Krnl a MemRef from within another MemRef starting at a specific offset.";
+  let description = [{
+    Retreieves a MemRef from within another MemRef.
```

Comment:
> I am a bit concerned when everything we do is with constant sizes... it will make the support of dyn memory harder. I let you judge if we should charge ahead with this PR first and then add it, or if we should add it from the start.
> 
> Otherwise, please add comments to the op definitions.

For now we only support static sizes but dynamic support can be added. I recommend adding dynamic support in a separate patch since this patch is already big enough.

Reply:
There's no restriction as to what types you can use for the input memref. The offset is always going to be in the unit of the first argument. it is up to the "user" of the krnl.getref instruction to set the offset to be a valid offset within the first MemRef argument.

In our case what happens is that we have a transformation which takes in an alloc operation and transforms it into a correct alloc + getref instruction combination. We emit a flat MemRef of i8 type and an offset to match.

Yes the offset can be more than just a constant but we don't yet have a use case for that. In order to emit the offset as not a constant we would have to teach onnx-mlir to emit an expression for this offset. When that happens it will be fine to pass it to the getref instruction.